libs/utils/src/http/error.rs

```diff
@@ -15,24 +15,32 @@ pub enum ApiError {
     Unauthorized(String),
 
     #[error("NotFound: {0}")]
-    NotFound(String),
+    NotFound(anyhow::Error),
 
     #[error("Conflict: {0}")]
     Conflict(String),
 
     #[error(transparent)]
-    InternalServerError(#[from] anyhow::Error),
+    InternalServerError(anyhow::Error),
 }
 
 impl ApiError {
-    pub fn from_err<E: Into<anyhow::Error>>(err: E) -> Self {
+    pub fn from_internal_err<E: Into<anyhow::Error>>(err: E) -> Self {
         Self::InternalServerError(anyhow!(err))
     }
 
+    /// Creates an `ApiError::BadRequest` by converting it into an `anyhow::Error`
+    ///
+    /// If the type is already an `anyhow::Error`, directly using `BadRequest` enum variant should
+    /// be preferred.
+    pub fn from_bad_request<E: Into<anyhow::Error>>(err: E) -> Self {
+        Self::BadRequest(err.into())
+    }
```

Comment:
This is a bit confusing. From the function name and comment, I though that this converts *from* BadRequest *into* `anyhow:Error`.

Reply:
Makes sense - I decided to remove both `from_bad_request` and `from_internal_err` so that the way errors are being converted can be a little bit more explicit. To that end, I've also added explicit types to the places where possible tokio task joining errors are handled - for example:

https://github.com/neondatabase/neon/blob/0b53c20685fe82fd35b8cbd5e866acfe0d6161bf/pageserver/src/http/routes.rs#L204-L210

This might be a _bit_ much tbh; I'm not sure where we want the balance of writability vs readability/verboseness to be, but I thought I'd just go for it.